crates/common/tedge_config/src/tedge_config_cli/error.rs

```diff
@@ -18,6 +18,9 @@ pub enum TEdgeConfigError {
     #[error("Config file not found: {0}")]
     ConfigFileNotFound(std::path::PathBuf),
 
+    #[error("Config directory not found: {0}")]
+    ConfigDirNotFound(std::path::PathBuf),
+
```

Comment:
This is not a `tedge_config` error but an error raised by the `config_plugin`. It would be more appropriate to move this error case there.

Reply:
The below function ` pub fn from_tedge_config(
        config_dir: impl AsRef<Path>,
        tedge_config: &TEdgeConfig,
    ) -> Result<ConfigManagerConfig, TEdgeConfigError> ` returns only `TEdgeConfigError`, so thought this is the right place. And we are validating the path inside this function.